### DISCUSSION TITLE
Column feature LP Age

### DISCUSSION THREAD

**Discussion by BenjaDomb** [ds4dm/ecole#257] [Questions] [answered]
Hello,
I don't understand the meaning of the LP Age feature and how is calculated. I'll appreciate it if someone could explain it to me.

Thank you

**Comment by BenjaDomb** [comment]
Please, someone than can help me understand this feature

**Comment by gasse** [comment, marked as answer]
Hi @benjadb ,

I'm not sure which feature your refer to. Are you interested in the `scaled_age` feature from the `NodeBipartite` observation function ? You can find how this feature is calculated in Ecole, for both LP columns and rows, here:
https://github.com/ds4dm/ecole/blob/e9522ef138cd52de3859b15433b02e07cd43f2b2/libecole/src/observation/nodebipartite.cpp#L141
https://github.com/ds4dm/ecole/blob/e9522ef138cd52de3859b15433b02e07cd43f2b2/libecole/src/observation/nodebipartite.cpp#L241

You can also look at SCIP's API for details about what the LP age means:
https://www.scipopt.org/doc/html/group__PublicRowMethods.php#gaf33c0582cf3c27fb12d93efdfb9b9156
https://www.scipopt.org/doc/html/group__PublicColumnMethods.php#ga417a303f33362d817f10602803511966

There it says:
> gets the age of a column, i.e., the total number of successive times a column was in the LP and was 0.0 in the solution

Hope that helps.

Best,
Maxime

**Reply by BenjaDomb** [reply]
I understood perfectly. Thank you!
